src/main/scala/scala/swing/Publisher.scala

```diff
@@ -162,11 +162,11 @@ abstract class RefBuffer[A <: AnyRef] extends mutable.Buffer[A] with SingleRefCo
   protected[this] def removeReference(ref: Reference[A]): Unit = { underlying -= ref }
 }
 
-private[swing] abstract class RefSet[A <: AnyRef] extends mutable.Set[A] with SingleRefCollection[A] { self =>
+private[swing] abstract class RefSet[A <: AnyRef] extends mutable.Set[A] with MutableSetShim[A] with SingleRefCollection[A] { self =>
```

Comment:
Do you know what `purgeReferences` does? I would assume that we cannot just use `clear` as defined by `MutableSetShim`, but would rather ensure that `purgeReferences` is called as well?

Reply:
I haven't looked at it.

Reply:
Actually, RefBuffer is not used anywhere in the code base, lol